artifactory.py

```diff
@@ -2640,17 +2642,15 @@ def promote_build(
         :param fail_fast: fail and abort the operation upon receiving an error. Default: true
         :return:
         """
-        url = f"/api/build/promote/{build_name}/{build_number}"
+        url = f"{self.drive}/api/build/promote/{build_name}/{build_number}"
 
         if not isinstance(properties, dict):
             raise ArtifactoryException("properties must be a dict")
 
-        iso_time = datetime.datetime.now().astimezone().isoformat()
-        params = {
+        json_data = {
             "status": status,
             "comment": comment,
             "ciUser": ci_user,
-            "timestamp": iso_time,
```

Comment:
What is jfrog api guide tells about the var?
Can we fix it instead of deleting?

Reply:
I tried to fix it, but failed.

Reply:
By the way, even if the `timestamp ` parameter is not added to the request body, the backend will automatically add the `timestamp ` parameter.

Reply:
What is the error that you see?
It should accept ISO timestamp

Reply:
When I first run, the error as follows:
```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/dohq_artifactory/exception.py", line 22, in raise_for_status
    response.raise_for_status()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/requests-2.27.1-py3.7.egg/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error:  for url: https://xxx.jfrog.com/artifactory/api/build/promote/demo/13

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/donghui/Desktop/repos/pylib-scripts/test_art.py", line 16, in <module>
    arti_build.promote_build(build_name=name, build_number=number, target_repo="maven-releases", properties={}, ci_user='jenkins')
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/artifactory.py", line 2686, in promote_build
    timeout=self.timeout,
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/artifactory.py", line 725, in rest_post
    raise_for_status(response)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/dohq_artifactory/exception.py", line 41, in raise_for_status
    raise ArtifactoryException(error_info_dict["message"]) from exception
dohq_artifactory.exception.ArtifactoryException: com.fasterxml.jackson.databind.JsonMappingException: java.text.ParseException: Unparseable date: "2022-08-03T23:51:05.145+08:00" (through reference chain: org.jfrog.build.api.Build["statuses"]->java.util.ArrayList[9]->org.jfrog.build.api.release.PromotionStatus["timestampDate"])
```
Then I run again, the error as follows, and I noticed that the build wad deleted.
```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/dohq_artifactory/exception.py", line 22, in raise_for_status
    response.raise_for_status()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/requests-2.27.1-py3.7.egg/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error:  for url: https://xxx.jfrog.com/artifactory/api/build/promote/demo/13

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/donghui/Desktop/repos/pylib-scripts/test_art.py", line 16, in <module>
    arti_build.promote_build(build_name=name, build_number=number, target_repo="maven-releases", properties={}, ci_user='jenkins')
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/artifactory.py", line 2686, in promote_build
    timeout=self.timeout,
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/artifactory.py", line 725, in rest_post
    raise_for_status(response)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/dohq_artifactory/exception.py", line 41, in raise_for_status
    raise ArtifactoryException(error_info_dict["message"]) from exception
dohq_artifactory.exception.ArtifactoryException: Cannot find builds by the name 'demo' and the number '13' and buildRepo 'artifactory-build-info'.
```

Reply:
@beliaev-maksim your thoughts?

Reply:
I think we need to try first with some "hardcoded" value of timestamp and see if the error is still produced. Let's pick some ISO timestamp from internet and try it out.
then let's configure code the way, that it will produce proper format

Reply:
I updated the code and solved the problem of time format.